contracts/PoolManager.sol

```diff
@@ -49,9 +50,17 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
     /// @inheritdoc IPoolManager
     mapping(Currency currency => uint256) public override reservesOf;
 
+    /// @inheritdoc IPoolManager
+    WETH public immutable WETH_CONTRACT;
+
+    Currency public immutable WRAPPED_NATIVE;
+
     mapping(PoolId id => Pool.State) public pools;
 
-    constructor(uint256 controllerGasLimit) Fees(controllerGasLimit) ERC1155("") {}
+    constructor(uint256 controllerGasLimit, WETH wrappedNative) Fees(controllerGasLimit) ERC1155("") {
+        WETH_CONTRACT = wrappedNative;
+        WRAPPED_NATIVE = Currency.wrap(address(WETH_CONTRACT));
```

Comment:
Are you saving WRAPPED_NATIVE as an immutable bc there is some gas cost overhead to wrap? Or another reason?